build-aux/m4/bitcoin_runtime_lib.m4

```diff
@@ -0,0 +1,39 @@
+# On some platforms clang builtin implementations
+# require compiler-rt as a runtime library to use.
+#
+# See:
+# - https://bugs.llvm.org/show_bug.cgi?id=28629
+
+m4_define([_CHECK_RUNTIME_testbody], [[
+  #if defined(__clang__) && defined(__has_builtin)
+  #if __has_builtin(__builtin_mul_overflow)
+  bool f(long long x, long long y, long long* p)
+  {
+    return __builtin_mul_overflow(x, y, p);
+  }
+  #endif
+  #endif
+
+  int main() { return 0; }
+]])
+
+AC_DEFUN([CHECK_RUNTIME_LIB], [
+
+  AC_LANG_PUSH(C++)
+
+  AC_MSG_CHECKING([whether __builtin_mul_overflow can be used without compiler-rt])
```

Comment:
AIUI, this test succeeds with non-clang, but `__builtin_mul_overflow` doesn't actually get checked at all. Maybe the test itself should be conditional on using clang?